extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/FastBootHibernateReactivePersistenceProvider.java

```diff
@@ -82,10 +84,19 @@ public EntityManagerFactory createEntityManagerFactory(String emName, Map proper
         if (properties == null)
             properties = new HashMap<Object, Object>();
         // These are pre-parsed during image generation:
-        final List<QuarkusPersistenceUnitDescriptor> units = PersistenceUnitsHolder.getPersistenceUnitDescriptors();
 
-        for (PersistenceUnitDescriptor unit : units) {
-            //if the provider is not set, don't use it as people might want to use Hibernate ORM
+        // We might have multiple persistence unit descriptors of different kinds when using reactive and orm together
+        // In this case this provider should take care only of reactive units
+        // Luckily we can return null in this API to use the next provider for the other one
+        // See jakarta.persistence.Persistence.createEntityManagerFactory(java.lang.String, java.util.Map)
+        final List<QuarkusPersistenceUnitDescriptor> units = PersistenceUnitsHolder
+                .getPersistenceUnitDescriptors()
+                .stream()
+                .filter(u -> u.getName().equals(emName))
+                .filter(u -> u.isReactive()) // we don't support orm units here
```

Comment:
I see that you filter by `isReactive()` here, so here at least there is no ambiguity even if the reactive and ORM PUs have the same name... So the problem of using the same name could possibly be limited to CDI?

Would be nice to get rid of the different name, because it's a cause of confusion in error messages. I guess it could be a follow-up PR though.